polyfill/lib/regex.mjs

```diff
@@ -12,12 +12,51 @@ const yearpart = /(?:[+-]\d{6}|\d{4})/;
 const monthpart = /(?:0[1-9]|1[0-2])/;
 const daypart = /(?:0[1-9]|[12]\d|3[01])/;
 export const datesplit = new RegExpCtor(
-  `(${yearpart.source})(?:-(${monthpart.source})-(${daypart.source})|(${monthpart.source})(${daypart.source}))`
+  Call(
+    ArrayPrototypeJoin,
+    [
+      `(?<yearpart>${yearpart.source})`,
+      `(?:-(?<monthpart>${monthpart.source})-`,
+      `(?<daypart>${daypart.source})|`,
+      `(?<monthpart>${monthpart.source})(?<daypart>${daypart.source}))`
+    ],
+    ['']
+  )
+);
+const sep = /:/;
+const hourminute = new RegExpCtor(`(?<hour>\\d{2})(?:(?:${sep.source})?(?<minute>\\d{2}))?`);
+const second2 = /(?<second>\d{2})/;
+const fraction = /(?:[.,](?<fraction>\d{1,9}))/;
+const secondsfraction = new RegExpCtor(`(?:${sep.source}?${second2.source}${fraction.source}?)`);
+const millismicrosfraction = /(?<millis>\d{2})(?:(?<micros>\d{2}))(?:[.,](?<nanos>\d{1,9}))?/;
```

Comment:
The names don't seem right here, it's 3 digits each of ms and µs and they should come after the decimal separator. Is it possible these capture groups should actually be referring to hours and minutes?